2.26.0 build failure (SDL_waylandmouse.c:324:21: error: variable 'mouse_buffer_listener' has initializer but incomplete type)

Hit this failure when packaging 2.26.0 for Gentoo.
```
libtool: compile:  x86_64-pc-linux-gnu-gcc -m32 -mfpmath=sse -O2 -pipe -march=native -fdiagnostics-color=always -frecord-gcc-switches -Wreturn-type -ggdb3 -DUSING_GENERATED_CONFIG_H -Iinclude -I/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/include -idirafter /var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/khronos -DSDL_BUILD_MAJOR_VERSION=2 -DSDL_BUILD_MINOR_VERSION=26 -DSDL_BUILD_MICRO_VERSION=0 -mmmx -msse -msse2 -Wall -fno-strict-aliasing -fvisibility=hidden -Wdeclaration-after-statement -Werror=declaration-after-statement -I/usr/include/pipewire-0.3 -I/usr/include/spa-0.2 -D_REENTRANT -D_REENTRANT -I/usr/lib/libffi/include -Igen -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -D_REENTRANT -DHAVE_LINUX_VERSION_H -MMD -MT build/SDL_waylandmouse.lo -c /var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c  -fPIC -DPIC -o build/.libs/SDL_waylandmouse.o
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:324:21: error: variable 'mouse_buffer_listener' has initializer but incomplete type
  324 | static const struct wl_buffer_listener mouse_buffer_listener = {
      |                     ^~~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:325:5: warning: excess elements in struct initializer
  325 |     mouse_buffer_release
      |     ^~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:325:5: note: (near initialization for 'mouse_buffer_listener')
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c: In function 'create_buffer_from_shm':
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:363:16: warning: implicit declaration of function 'wl_shm_create_pool' [-Wimplicit-function-declaration]
  363 |     shm_pool = wl_shm_create_pool(data->shm, shm_fd, size);
      |                ^~~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:363:14: warning: assignment to 'struct wl_shm_pool *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
  363 |     shm_pool = wl_shm_create_pool(data->shm, shm_fd, size);
      |              ^
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:364:17: warning: implicit declaration of function 'wl_shm_pool_create_buffer' [-Wimplicit-function-declaration]
  364 |     d->buffer = wl_shm_pool_create_buffer(shm_pool,
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:364:15: warning: assignment to 'struct wl_buffer *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
  364 |     d->buffer = wl_shm_pool_create_buffer(shm_pool,
      |               ^
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:370:5: warning: implicit declaration of function 'wl_buffer_add_listener'; did you mean 'wl_proxy_add_listener'? [-Wimplicit-function-declaration]
  370 |     wl_buffer_add_listener(d->buffer,
      |     ^~~~~~~~~~~~~~~~~~~~~~
      |     wl_proxy_add_listener
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:374:5: warning: implicit declaration of function 'wl_shm_pool_destroy'; did you mean 'wl_proxy_destroy'? [-Wimplicit-function-declaration]
  374 |     wl_shm_pool_destroy (shm_pool);
      |     ^~~~~~~~~~~~~~~~~~~
      |     wl_proxy_destroy
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c: In function 'Wayland_CreateCursor':
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:401:37: error: 'WL_SHM_FORMAT_ARGB8888' undeclared (first use in this function); did you mean 'SDL_PIXELFORMAT_ARGB8888'?
  401 |                                     WL_SHM_FORMAT_ARGB8888) < 0)
      |                                     ^~~~~~~~~~~~~~~~~~~~~~
      |                                     SDL_PIXELFORMAT_ARGB8888
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:401:37: note: each undeclared identifier is reported only once for each function it appears in
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:413:25: warning: implicit declaration of function 'wl_compositor_create_surface' [-Wimplicit-function-declaration]
  413 |         data->surface = wl_compositor_create_surface(wd->compositor);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:413:23: warning: assignment to 'struct wl_surface *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
  413 |         data->surface = wl_compositor_create_surface(wd->compositor);
      |                       ^
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:414:9: warning: implicit declaration of function 'wl_surface_set_user_data'; did you mean 'wl_proxy_set_user_data'? [-Wimplicit-function-declaration]
  414 |         wl_surface_set_user_data(data->surface, NULL);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
      |         wl_proxy_set_user_data
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c: In function 'Wayland_CreateSystemCursor':
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:443:24: warning: assignment to 'struct wl_surface *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
  443 |         cdata->surface = wl_compositor_create_surface(data->compositor);
      |                        ^
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c: In function 'Wayland_FreeCursorData':
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:468:13: warning: implicit declaration of function 'wl_buffer_destroy' [-Wimplicit-function-declaration]
  468 |             wl_buffer_destroy(d->buffer);
      |             ^~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:474:9: warning: implicit declaration of function 'wl_surface_destroy'; did you mean 'wl_proxy_destroy'? [-Wimplicit-function-declaration]
  474 |         wl_surface_destroy(d->surface);
      |         ^~~~~~~~~~~~~~~~~~
      |         wl_proxy_destroy
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c: In function 'Wayland_ShowCursor':
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:521:9: warning: implicit declaration of function 'wl_surface_set_buffer_scale' [-Wimplicit-function-declaratio ]
  521 |         wl_surface_set_buffer_scale(data->surface, scale);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:522:9: warning: implicit declaration of function 'wl_pointer_set_cursor' [-Wimplicit-function-declaration]
  522 |         wl_pointer_set_cursor(pointer,
      |         ^~~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:527:9: warning: implicit declaration of function 'wl_surface_attach' [-Wimplicit-function-declaration]
  527 |         wl_surface_attach(data->surface, data->buffer, 0, 0);
      |         ^~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:528:9: warning: implicit declaration of function 'wl_surface_damage' [-Wimplicit-function-declaration]
  528 |         wl_surface_damage(data->surface, 0, 0, data->w, data->h);
      |         ^~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:529:9: warning: implicit declaration of function 'wl_surface_commit' [-Wimplicit-function-declaration]
  529 |         wl_surface_commit(data->surface);
      |         ^~~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c: At top level:
/var/tmp/portage/media-libs/libsdl2-2.26.0/work/SDL2-2.26.0/src/video/wayland/SDL_waylandmouse.c:324:40: error: storage size of 'mouse_buffer_listener' isn't known
  324 | static const struct wl_buffer_listener mouse_buffer_listener = {
      |                                        ^~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:754: build/SDL_waylandmouse.lo] Error 1
 * ERROR: media-libs/libsdl2-2.26.0::gentoo failed (compile phase):
 *   emake failed
```

Build log: [build.log](https://github.com/libsdl-org/SDL/files/10072432/build.log)
config.log: [config.log](https://github.com/libsdl-org/SDL/files/10072435/config.log)

--- a/src/video/wayland/SDL_waylanddyn.h
+++ b/src/video/wayland/SDL_waylanddyn.h
@@ -166,6 +166,13 @@ void SDL_WAYLAND_UnloadSymbols(void);
 
 #else /* SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC */
 
+/*
+ * These must be included before libdecor.h, otherwise the libdecor header
+ * pulls in the system Wayland protocol headers instead of ours.
+ */
+#include "wayland-client-protocol.h"
+#include "wayland-egl.h"
+
 #ifdef HAVE_LIBDECOR_H
 #include <libdecor.h>
 #endif

